proposals/0257-eliminate-vote-fees.md

```diff
@@ -0,0 +1,360 @@
+timely manner so that the cluster can reach consensus as fast as possible.
+Third, leaders should include votes that they receive as soon as possible for the
+same reason.
+
+Fourth, validator rewards should be linear in stake so that validators don't
+have an incentive to combine or split up to increase their rewards.
+
+Fifth, the total change in active stake cannot exceed 18% from epoch to epoch
+because this could result in a reversal of fast confirmations. The existing limit
+of 9% total stake activations + 9% total stake deactivations must be maintained
+for this security.
+
+Sixth, the votes must be limited to cap the size of the block and prevent DOS
+attacks on the network. Finally, the number of validators that can participate
+in consensus must be limited to protect Turbine operation.
+
+## Detailed Design
+
+### Vote base fees
+
+Vote base fees will be removed entirely.
+
+### Absolute Vote Credits
+
+Instead of distributing rewards based on relative performance as we do today,
+each validator in the active validator set will have a maximum number of inflationary
+rewards that they are eligble for. To determine this we first define the maximum
+stake weighted vote credits $M$ that they can earn in an epoch where they produce
+$L$ leader slots. Here $s_i$ is the validator's stake, and $S$ is the total stake.
+
+$$M = 12 s_i \times 432000 + 4 L S$$
+
+The first term represents the maximum voting credits $3/4$ of the TVC credits
+(16), and the second term is the maximum leader inclusion credits covered in the
+following section.
+
+Their rewards in each epoch will be based on the fraction of the stake weighted
+credits $c_{i}$ they actually earn relative to this maximum $M$ multiplied by the
+total inflation rewards available in the epoch $I$.
+
+$$ R_i = I \frac{c_i}{S M} $$
+
+Any unearned sol rewards will be burned.
+
+This represents a very slight reduction in total rewards available to validators:
+
+![rewards table](../assets/0257-rewardstbl.png)
+
+### Vote Inclusion Credits
+
+25% of issuance will be redirected from vote credits to vote inclusion credits.
+These credits will be awarded to validators based on the number of votes they
+include in a timely manner. Each vote inclusion credit is awarded to the leader
+and is worth 1/3 of the vote credits awarded to the validator that the vote belongs
+to.
+
+For example assume that a latency 1 vote comes in from voter $v$ to be included
+in leader $l$'s block. $v$ will earn 12 credits based on their stake weight, and
+$l$ will earn 4 credits based on $v$'s stake weight:
+
+$c_v := c_v + 12 s_v$
+
+$c_l := c_l + 4 s_v$
+
+Note that this accounting matches the $M$ derived above, if $v$ votes perfectly
+and includes everyones votes (including their own):
+
+$$c_v = 12 s_v \times 432000 + 4 L \sum_{i\in V} s_i = M$$
+
+In order to implement this change, the vote program will track credits $c_v$ as
+stake weighted totals.
```

Comment:
could consider also incrementing on vote rather than root to eliminate lockout expiration mods as well but might be too much for this proposal